pkg/operator2/starter.go

```diff
@@ -163,83 +211,136 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 
-	for _, informer := range []interface {
-		Start(stopCh <-chan struct{})
-	}{
-		kubeInformersNamespaced,
-		kubeSystemNamespaceInformers,
-		authOperatorConfigInformers,
-		routeInformersNamespaced,
-		configInformers,
-		resourceSyncerInformers,
-	} {
-		informer.Start(ctx.Done())
-	}
+	operatorCtx.informersToRunFunc = append(operatorCtx.informersToRunFunc, routeInformersNamespaced.Start, configInformers.Start, kubeSystemNamespaceInformers.Start)
+
+	operatorCtx.controllersToRunFunc = append(
+		operatorCtx.controllersToRunFunc,
+		clusterOperatorStatus.Run,
+		configOverridesController.Run,
+		logLevelController.Run,
+		routerCertsController.Run,
+		managementStateController.Run,
+		func(ctx context.Context, workers int) { staleConditions.Run(ctx, workers) },
+		func(ctx context.Context, workers int) { ingressStateController.Run(workers, ctx.Done()) },
+		func(ctx context.Context, _ int) { operator.Run(ctx.Done()) })
+
+	return nil
+}
+
+func prepareOauthAPIServerOperator(controllerContext *controllercmd.ControllerContext, operatorCtx *operatorContext) error {
+	eventRecorder := controllerContext.EventRecorder.ForComponent("oauth-apiserver")
 
-	for _, controller := range []interface {
-		Run(ctx context.Context, workers int)
-	}{
-		resourceSyncer,
-		clusterOperatorStatus,
-		configOverridesController,
-		logLevelController,
-		routerCertsController,
-		managementStateController,
-	} {
-		go controller.Run(ctx, 1)
+	// add syncing for etcd certs for oauthapi-server
+	if err := operatorCtx.resourceSyncController.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Namespace: "openshift-oauth-apiserver", Name: "etcd-serving-ca"},
+		resourcesynccontroller.ResourceLocation{Namespace: "openshift-config", Name: "etcd-serving-ca"},
+	); err != nil {
+		return err
+	}
+	if err := operatorCtx.resourceSyncController.SyncSecret(
+		resourcesynccontroller.ResourceLocation{Namespace: "openshift-oauth-apiserver", Name: "etcd-client"},
+		resourcesynccontroller.ResourceLocation{Namespace: "openshift-config", Name: "etcd-client"},
+	); err != nil {
+		return err
 	}
 
-	go operator.Run(ctx.Done())
-	go staleConditions.Run(1, ctx.Done())
-	go ingressStateController.Run(1, ctx.Done())
+	authAPIServerWorkload := workload.NewOAuthAPIServerWorkload(
+		operatorCtx.operatorClient.Client,
+		workloadcontroller.CountNodesFuncWrapper(operatorCtx.kubeInformersForNamespaces.InformersFor("").Core().V1().Nodes().Lister()),
+		workloadcontroller.EnsureAtMostOnePodPerNode,
+		"openshift-oauth-apiserver",
+		os.Getenv("IMAGE_OAUTH_APISERVER"),
+		os.Getenv("OPERATOR_IMAGE"),
+		operatorCtx.kubeClient,
+		eventRecorder,
+		operatorCtx.versionRecorder)
+
+	apiServerControllers, err := apiservercontrollerset.NewAPIServerControllerSet(
+		operatorCtx.operatorClient,
+		eventRecorder,
+	).WithWorkloadController(
+		"OAuthAPIServerController",
+		"openshift-authentication-operator",
+		"openshift-oauth-apiserver",
+		os.Getenv("OPERATOR_IMAGE_VERSION"),
```

Comment:
this is used to set the version of our new component, not sure if we have to introduce a new env var.